packages/react-server/src/ReactServerStreamConfigBun.js

```diff
@@ -22,7 +22,7 @@ export opaque type Chunk = string;
 export type BinaryChunk = $ArrayBufferView;
 
 export function scheduleWork(callback: () => void) {
-  callback();
+  setImmediate(callback);
```

Comment:
We kind of should add another one here. Basically, you can use `setImmediate`, but it only runs after all the other tasks are complete - and that's tasks, not microtasks. 

https://github.com/oven-sh/bun/blob/7f1880cafb21628fdfda5387ff897a9693845ead/src/bun.js/event_loop.zig#L1340-L1378

What if you did `process.nextTick`? Microtasks are drained after each one of those, and unlike `setImmediate` it doesn't have a dependency on the (async) event loop.

Reply:
interesting, my understanding is Node schedules `setImmediate` and `setTimeout(f, 0)` simiarly but we use `setImmediate` because it isn't throttled like setTimeout may be. In Bun it seems like `setImmediate` is actually lower priority than `setTimeout(f, 0)`. We can use that for now however if it is also throttled that's not great either since it puts an upper bound on iterations and it unnecessarily delays work start.

We can't use microtasks here because we are going to add a new scheduling primitive for microtasks and we need to be able to ensure some work is scheduled in a macrotask to ensure the microtask work is flushed first.

Would postMessage be better? that's what we're going to use for the browser

Reply:
Microtasks are drained after each process.nextTick runs in both node & bun. 

The code is essentially  (with some try catch and promise rejection handling and async loca storage) 

```js
while (tick = next()) {
  tick();
  drainMicrotasks()
}
```

I think it’s likely you want nextTick in bun and node, or to override nextTick to make yours have higher priority

Reply:
Well actually we want it to be a lower priority. If two fetches are resolving we'd prefer if both resolved before we restarted work otherwise we might stall on the second one. So we want a low priority task, we just don't want it to be artificially delayed. I maybe misinterpreted your earlier comment but my understanding is that in Node setImmediate will usually run before setTimeout(..., 0). In Bun is this true too? I wouldn't want it to run later than setTimeout(,0) but I also don't want it to run before IO tasks have had a chance to run which should be prioritized. The concern with setTimeout is that if there are many they may be throttled in some environments